File: src/Nebula/Modules/Account/Models/Warehouse.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Nebula.Modules.Account.Models;

public class Warehouse
{
    [Key]
    public Guid Id { get; set; } = Guid.NewGuid();

    /// <summary>
    /// Identificador de la empresa al que pertenece.
    /// </summary>
    public Guid? CompanyId { get; set; } = null;

    [ForeignKey(nameof(CompanyId))]
    public Company Company { get; set; } = new Company();

    /// <summary>
    /// Nombre Almacén.
    /// </summary>
    public string Name { get; set; } = string.Empty;

    /// <summary>
    /// Observación.
    /// </summary>
    public string Remark { get; set; } = string.Empty;
}

File: ngx-angular/src/app/account/company/interfaces/warehouse.ts
export class Warehouse {
  constructor(
    public id: string | any = undefined,
    public companyId: string = "",
    public name: string = "",
    public remark: string = "") {
  }
}

export class WarehouseDataModal {
  constructor(
    public title: string = "",
    public type: "ADD" | "EDIT" = "ADD",
    public warehouse: Warehouse = new Warehouse()) {
  }
}
